### PR TITLE
Increase hardcoded acceleration limit for simulation

### PR DESCRIPTION
**Pull request by MishkaMN**
# PR Details
## Description
Increasing the hardcoded acceleration limit so that we can test more range of motion in simulation. 

## Related GitHub Issue
N/A

## Related Jira Key
https://usdot-carma.atlassian.net/browse/CDAR-825

## Motivation and Context
vru use case 1 verification testing prep

## How Has This Been Tested?
sim pc 1

## Types of changes

- [ ] Defect fix (non-breaking change that fixes an issue)
- [X] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [X] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING**](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) document.
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed.

### CODE REVIEW

File: common/hardcoded_params/include/hardcoded_params/control_limits/control_limits.h [hunk 1]
```diff
@@ -21,9 +21,19 @@ namespace hardcoded_params
...
- * The maximum allowable longitudinal velocity of the vehicle in m/s 
+ * The maximum allowable longitudinal velocity of the vehicle in m/s
  */
 constexpr double MAX_LONGITUDINAL_VELOCITY_MPS = 35.7632;
 constexpr double MAX_LONGITUDINAL_ACCEL_MPS2 = 3.5;
+
+/**
+* The maximum allowable simulation parameters
+* At the time of writing this, platform uses CARLA 0.9.10 as the vehicle's physics simulator.
+* CARLA's this version uses below hardcoded value for its vehicles (so even if increasing this value,
```
Comment (JonSmet) [inline common/hardcoded_params/include/hardcoded_params/control_limits/control_limits.h]: Minor typo: Maybe change `CARLA's this version` to `CARLA 0.9.10`?

File: core_planning/twist_filter/src/twist_filter.cpp [hunk 2]
```diff
@@ -53,17 +53,34 @@ carma_ros2_utils::CallbackReturn TwistFilter::handle_on_configure(const rclcpp_l
...
+  bool use_sim_time = false;
+  get_parameter<bool>("use_sim_time", use_sim_time);
+
+  auto hardcoded_acceleration_limit = MAX_LONGITUDINAL_ACCEL_HARDCODED_LIMIT_M_S_2;
+
+  // if simulation
+  if (use_sim_time)
+  {
+    hardcoded_acceleration_limit = MAX_SIMULATION_LONGITUDINAL_ACCEL_HARDCODED_LIMIT_M_S_2;
+    RCLCPP_INFO_STREAM(rclcpp::get_logger("logger"), "Simulation mode detected. Accounting hardcoded limit longitudinal acceleration limit used: " << hardcoded_acceleration_limit);
```
Comment (JonSmet) [inline core_planning/twist_filter/src/twist_filter.cpp]: Are we able to use `rclcpp::get_logger()` here?

File: core_planning/twist_filter/src/twist_filter.cpp [hunk 3]
```diff
@@ -53,17 +53,34 @@ carma_ros2_utils::CallbackReturn TwistFilter::handle_on_configure(const rclcpp_l
...
+  }
+  else
+  {
+    RCLCPP_INFO_STREAM(rclcpp::get_logger("logger"), "Accounting hardcoded limit, longitudinal acceleration limit used: " << hardcoded_acceleration_limit);
```
Comment (JonSmet) [inline core_planning/twist_filter/src/twist_filter.cpp]: Are we able to use `rclcpp::get_logger()` here?

File: core_planning/twist_filter/src/twist_filter.cpp (hunk 2 above)
Comment (JonSmet) [inline core_planning/twist_filter/src/twist_filter.cpp]: Minor typo: can we change "Accounting hardcoded limit" to "Accounting for hardcoded limit,"

File: core_planning/twist_filter/src/twist_filter.cpp (hunk 3 above)
Comment (JonSmet) [inline core_planning/twist_filter/src/twist_filter.cpp]: Minor typo: can we change "Accounting hardcoded limit" to "Accounting for hardcoded limit"

**Review by JonSmet — CHANGES_REQUESTED** [review]

File: common/hardcoded_params/include/hardcoded_params/control_limits/control_limits.h (hunk 1 above)
Comment (MishkaMN) [reply common/hardcoded_params/include/hardcoded_params/control_limits/control_limits.h]: Edited

File: core_planning/twist_filter/src/twist_filter.cpp (hunk 2 above)
Comment (MishkaMN) [reply core_planning/twist_filter/src/twist_filter.cpp]: replaced with get_logger()

File: core_planning/twist_filter/src/twist_filter.cpp (hunk 3 above)
Comment (MishkaMN) [reply core_planning/twist_filter/src/twist_filter.cpp]: replaced with get_logger()

File: core_planning/twist_filter/src/twist_filter.cpp (hunk 2 above)
Comment (MishkaMN) [reply core_planning/twist_filter/src/twist_filter.cpp]: reworded

File: core_planning/twist_filter/src/twist_filter.cpp (hunk 3 above)
Comment (MishkaMN) [reply core_planning/twist_filter/src/twist_filter.cpp]: reworded

**Review by JonSmet — APPROVED** [review]

**Pull request merged**